src/OpenTelemetry/Instrumentation/PropertyFetcher.cs

```diff
@@ -39,12 +39,19 @@ public PropertyFetcher(string propertyName)
         }
 
         /// <summary>
-        /// Fetch the property from the object.
+        /// Try to fetch the property from the object.
         /// </summary>
         /// <param name="obj">Object to be fetched.</param>
-        /// <returns>Property fetched.</returns>
-        public T Fetch(object obj)
```

Comment:
Hmm... I was thinking that this would stay for usages where you don't care and the TryFetch would be added. But I guess it's bad practice to not know if the operation succeeded or not. Just makes the call site for all the usages of this a little uglier.

Reply:
for users who doesn't care if that works or not, they will just use it and use the out variable. But, we shouldn't throw errors to the user. so, that's the overall idea :)

Reply:
Agreed. Good call. :-)

Reply:
If we want to keep it around, we could do:
```csharp
        public T Fetch(object obj)
        {
            if (!this.TryFetch(obj, out T value))
            {
                throw new ArgumentException("Supplied object was null or did not match the expected type.", nameof(obj));
            }

            return value;
        }
```

The idea was to remove the old one because it had hidden behaviors that could lead to errors. This version basically makes the hidden behaviors explicit.

Reply:
should i add or keep just with tryfetch?

Reply:
@CodeBlanch I like that. I think, like you've said before, this is working for most use cases without issues. So I kind of like the simpler call sites.

Reply:
let me add then 👍

Reply:
just added!